cli/helpers/exec.go

```diff
@@ -76,3 +83,31 @@ func CreateTmpFile(contents string) (string, error) {
 
 	return tmpfile.Name(), nil
 }
+
+// Kubectl invoces the `kubectl` command in PATH, running the specified command.
+// It returns the command output and/or error.
+func Kubectl(command string) (string, error) {
+	_, err := exec.LookPath("kubectl")
+	if err != nil {
+		return "", errors.Wrap(err, "kubectl not in path")
+	}
+
+	currentdir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	cmd := fmt.Sprintf("kubectl " + command)
+
+	return RunProc(cmd, currentdir, false)
+}
+
+func SpinnerWaitCommand(message string, funk ExternalCommandFunc) (string, error) {
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond) // Build our new spinner
+	s.Start()                                                    // Start the spinner
+	defer s.Stop()
+
+	s.Suffix = emoji.Sprintf(" %s :zzz:", message)
```

Comment:
Does it make sense to move this statement to just before `s.Start()` above ?

In the position it is in it feels as if the main goroutine accesses the data structures of the goroutine inside of the spinner, i.e. changing the suffix underneath it. Before `s.Start()` it would make the change before the spinner's goroutine is active.

Hm ... Will activate the `-race` option for building he app, see what the data race detectors tells me.

Reply:
Yup. Immediate data race involving the spinner.
Now to check if my proposed change will fix it.

```
ARNING: DATA RACE
Read at 0x00c00020a718 by goroutine 27:
  github.com/briandowns/spinner.(*Spinner).Start.func1()
      /home/work/SUSE/go/pkg/mod/github.com/briandowns/spinner@v1.12.0/spinner.go:307 +0x339

Previous write at 0x00c00020a718 by main goroutine:
  github.com/suse/carrier/cli/kubernetes.(*Cluster).WaitForPodBySelectorRunning()
      /home/work/SUSE/dev/carrier/cli/kubernetes/cluster.go:178 +0x229
  github.com/suse/carrier/cli/deployments.Quarks.apply()
      /home/work/SUSE/dev/carrier/cli/deployments/quarks.go:70 +0x429
  github.com/suse/carrier/cli/deployments.Quarks.Deploy()
      /home/work/SUSE/dev/carrier/cli/deployments/quarks.go:95 +0x2f5
  github.com/suse/carrier/cli/deployments.(*Quarks).Deploy()
      <autogenerated>:1 +0xd7
  github.com/suse/carrier/cli/kubernetes.(*Installer).Install()
      /home/work/SUSE/dev/carrier/cli/kubernetes/installer.go:96 +0x1d8
  github.com/suse/carrier/cli/cmd.Install()
      /home/work/SUSE/dev/carrier/cli/cmd/install.go:103 +0x38a
  github.com/spf13/cobra.(*Command).execute()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x8e0
  github.com/spf13/cobra.(*Command).ExecuteC()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x499
  github.com/suse/carrier/cli/cmd.Execute()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895 +0x2c5
  main.main()
      /home/work/SUSE/dev/carrier/cli/main.go:6 +0x2f

Goroutine 27 (running) created at:
  github.com/briandowns/spinner.(*Spinner).Start()
      /home/work/SUSE/go/pkg/mod/github.com/briandowns/spinner@v1.12.0/spinner.go:281 +0xef
  github.com/suse/carrier/cli/kubernetes.(*Cluster).WaitForPodBySelectorRunning()
      /home/work/SUSE/dev/carrier/cli/kubernetes/cluster.go:176 +0xee
[...]
```

Reply:
No, apparently not. Hm. That means the race seen above is not what I thought. Looking deeper now instead of assuming.

Reply:
Ah! The spinner creations in the cluster.go have the same ordering issue on `.Suffix`. 'k, change these too.

Reply:
The one in the loop over the pods (in `WaitForPodBySelectorRunning`) requires a full cycle of "stop; modify; start". ... Pity that the spinner package does not have a method change the suffix mid operation. That could have used the internal lock employed by the spinner.

And with that no races in this area are seen any longer.
Will make a PR for this now.

Reply:
PR #44

Reply:
Merged. Resolving.

Reply:
Merged, thanks!